xsuite/src/world/signer.ts

```diff
@@ -56,26 +56,30 @@ export class KeystoreSigner extends UserSigner {
 export class Keystore {
   data: any;
   password: string;
+  kind: "mnemonic" | "secretKey";
 
   constructor(data: any, password: string) {
     this.data = data;
     this.password = password;
+    this.kind = this.getSupportedKeystoreKind();
```

Comment:
Should rather be a simple function that takes the `data` as parameter. Would make the code more understandable.

Reply:
I don't know if you meant a static class function or a function outside of the class.
For now I implemented the first option.